File: Programmers/시저 암호.cpp
#include <string>
#include <iostream>
#include <vector>

using namespace std;

string solution(string s, int n) {
	for (int i = 0; i < s.size(); i++)
	{
		int tmp = n;
		if (s[i] == 32) continue;
		if (s[i] > 64 && s[i] < 91)
		{
			if (s[i] + n > 90)
				tmp -= 26;
		}
		else {
			if (s[i] + n > 122)
				tmp -= 26;
		}
		s[i] += tmp;
	}
	return s;
}

int main() {
	string s;
	int n;
	cin >> s >> n;
	cout << solution(s, n);
	system("pause");
}

File: Programmers/시저 암호.java
class Solution {
    public String solution(String s, int n) {
        String answer = "";
        char [] tmp = s.toCharArray();
        for(int i = 0; i < tmp.length; i++) {
            if(tmp[i] >= 65 && tmp[i] <=90){
                if((tmp[i] + n)/91 == 0)
                    answer += (char)(tmp[i] + n);
                else
                    answer += (char)((tmp[i] + n)%90 + 64);
                }
            else if(tmp[i] == ' ')
                answer += ' ';
            else
            {
                if((tmp[i] + n)/123 == 0)
                    answer += (char)(tmp[i] + n);
                else
                    answer += (char)((tmp[i] + n)%122 + 96);
            }
        }
        return answer;
    }
}